lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -2532,7 +2535,16 @@ static LogicalResult zeroRepeat(StaticRepeatOp op, PatternRewriter &rewriter) {
 void StaticRepeatOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
                                                  MLIRContext *context) {
   patterns.add(emptyControl<StaticRepeatOp>);
-  patterns.add(zeroRepeat);
+  patterns.add(zeroRepeat<StaticRepeatOp>);
```

Comment:
I think the better pattern here is to have a simplification for (`RepeatOp` with statically known `count`) -> (`StaticRepeatOp`), then this canonicalization pattern will essentially handle both.